typescript/examples/langchain-solana-chatbot/chatbot.ts

```diff
@@ -0,0 +1,262 @@
+import { AgentKit, SolanaKeypairWalletProvider, walletActionProvider } from "@coinbase/agentkit";
+import { getLangChainTools } from "@coinbase/agentkit-langchain";
+import { HumanMessage } from "@langchain/core/messages";
+import { MemorySaver } from "@langchain/langgraph";
+import { createReactAgent } from "@langchain/langgraph/prebuilt";
+import { ChatOpenAI } from "@langchain/openai";
+import { Keypair } from "@solana/web3.js";
+import bs58 from "bs58";
+import * as dotenv from "dotenv";
+import * as readline from "readline";
+
+dotenv.config();
+
+/**
+ * Validates that required environment variables are set
+ *
+ * @throws {Error} - If required environment variables are missing
+ * @returns {void}
+ */
+function validateEnvironment(): void {
+  const missingVars: string[] = [];
+
+  // Check required variables
+  const requiredVars = ["OPENAI_API_KEY", "SOLANA_RPC_URL"];
```

Comment:
is RPC URL the most appropriate way of determining which chain to use? We should use default RPCs IMO. We should also default to devnet for testing and quick onboarding.

Reply:
The way the `SolanaKeypairWalletProvider` is currently implemented does determine network based off of the RPC URL. The `SolanaKeypairWalletProvider` construction code creates a connection to that RPC URL, and requests the genesis hash from it. It uses this hash to determine which network it is on.

I think it's fine to require network and we pick the RPC from one of the selected networks. It would require a update to `SolanaKeypairWalletProvider`.

Reply:
Can we use https://github.com/coinbase/agentkit/blob/ba01f4c3400b54cd0af11052f896caf0c7573c94/typescript/agentkit/src/wallet-providers/solanaKeypairWalletProvider.ts#L70 to get the RPC URL for the network?

Reply:
Ah yes! Perfect. We'd just need to map the user-friendly network name to a cluster, which we have in our types, and then we can use that to get the RPC URLs

Reply:
I created a new PR to address this. After Chris' PR is merged, I will rebase and publish this PR to add it: https://github.com/coinbase/agentkit/pull/390